src/global/variables/Clause.java

```diff
@@ -1,5 +1,17 @@
 package global.variables;
 
-public interface Clause extends Regexable {
+public abstract class Clause implements Regexable {
```

Comment:
Did we need to do this?

Reply:
We needed the abstract class because we need a property that is common to all `Clause` objects, otherwise, we'd have to go add a `name` variable to every subclass of `Clause` to achieve the same effect.

Reply:
Oh I see, aye aye